nyxx/lib/src/internal/http/HttpHandler.dart

```diff
@@ -20,16 +24,41 @@ class _HttpHandler {
       return _handle(await this._noRateBucket._execute(request));
     }
 
-    // TODO: NNBD: try-catch in where
-    try {
-      final bucket = _buckets.firstWhere((element) => element.uri == request.uri);
+    final bucket = this._getBucketForRequest(request);
+    return _handle(await bucket._execute(request));
+  }
+
+  _HttpBucket _getBucketForRequest(_HttpRequest request) {
+    final reactionsRegexMatch = _bucketReactionsRegexp.firstMatch(request.uri.toString());
+    if (reactionsRegexMatch != null) {
+      final bucketMajorId = reactionsRegexMatch.group(1);
+      final bucketMessageId = reactionsRegexMatch.group(2);
+
+      return this._findBucketById("reactions/$bucketMajorId/$bucketMessageId");
```

Comment:
Why is it `reactions/`

Reply:
reactions have totaly different rate limtits then other routes so it needs to be in different bucket

Reply:
Ah right, no worries then :)